### PR TITLE
Try to make stork faster

### PR DESCRIPTION
**Pull request by parsing-science**
Our team has seen some issues where job libraries aren't getting updated properly. It's unclear if this PR fixes the issue, but it will make the code run faster, which should make it easier to debug.

The main change is that we use the workspace 2.0 API to list out the libraries in the prod folder rather than the libraries 1.2 API, which lists every single library in the entire workspace.

Pull Request Checklist
 - [x] Pull request includes a description of the change and the reason behind it
 - [x] CHANGELOG has been updated
 - [x] Version in `_version.py` has been updated
 - [x] README and docs have been updated (if applicable)
 - [ ] CI checks pass

### CODE REVIEW

File: tests/fixtures.py [hunk 1]
```diff
@@ -263,60 +250,57 @@ def library_8(prod_folder):
...
-            'name': 'test-library-1.0.3',
-            'folder': prod_folder,
-        },
-    ]
+def workspace_list_response(prod_folder):
+    workspace_list_response = {
+        'objects': [
+            {
+                'object_type': 'LIBRARY',
+                'path': '/PROD_FOLDER/awesome_library_b-4.2.3',
```
Comment (htorrence) [inline tests/fixtures.py:258]:
```suggestion
                'path': f'/{prod_folder}/awesome_library_b-4.2.3',
```

Comment (htorrence) [reply tests/fixtures.py:258]: I think this needs to get swapped to match the rest of the structure

**Review by htorrence — CHANGES_REQUESTED** [review]
A few small fixes, but the core looks like a great change. Thanks for working on this!

File: stork/update_databricks_library.py [hunk 2]
```diff
@@ -495,6 +356,7 @@ def update_databricks(logger, path, token, folder, update_jobs, cleanup):
                          ' to get set up')
     try:
         prod_folder = config.get(PROFILE, 'prod_folder')
+        prod_folder = '/Shared/DS_libraries'
```
[the remark this answers is not in the document]
Comment (htorrence) [reply stork/update_databricks_library.py:359]: I think this is causing the CI error

File: stork/file_name.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import re
+
```
[the remark this answers is not in the document]
Comment (parsing-science) [reply stork/file_name.py:2]: I just moved these to a separate file. I was having trouble following the code with so many functions in the same file. I can move it back if you want.

File: stork/update_databricks_library.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (parsing-science) [reply stork/update_databricks_library.py]: oof, good catch!

File: tests/fixtures.py (hunk 1 above)
Comment (parsing-science) [reply tests/fixtures.py:258]: Sure, I can change it. We don't actually use the path so that's why the tests still passed.

**Comment by parsing-science** [comment]
Back to you @htorrence !

File: stork/file_name.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (htorrence) [reply stork/file_name.py:2]: Nope, I'm fine having it in a separate file, just making sure I didn't need to check any changes here

File: tests/fixtures.py (hunk 1 above)
Comment (htorrence) [reply tests/fixtures.py]: Cool, I think I'd still rather have it consistent so it won't break in a weird way if someone uses the value in a future test

**Review by htorrence — APPROVED** [review]
LGTM!

**Pull request merged**